client/client.go

```diff
@@ -214,6 +187,48 @@ func (c *LogClient) UpdateRoot(ctx context.Context) error {
 	return nil
 }
 
+// WaitForInclusion blocks until the requested data has been verified with an inclusion proof.
+// This assumes that the data has already been submitted.
+// Best practice is to call this method with a context that will timeout.
+func (c *LogClient) WaitForInclusion(ctx context.Context, data []byte) error {
+	leaf := c.logVerifier.buildLeaf(data)
+
+	// Fetch the current Root to improve our chances at a valid inclusion proof.
+	if err := c.UpdateRoot(ctx); err != nil {
+		return err
+	}
+	if c.root.TreeSize == 0 {
+		// If the TreeSize is 0, wait for something to be in the log.
+		// It is illegal to ask for an inclusion proof with TreeSize = 0.
+		if err := c.waitForRootUpdate(ctx); err != nil {
+			return err
+		}
+	}
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			err := c.getInclusionProof(ctx, leaf.MerkleLeafHash, c.root.TreeSize)
+			s, ok := status.FromError(err)
+			if !ok {
+				return err
+			}
+			switch s.Code() {
+			case codes.OK:
+				return nil
+			case codes.NotFound:
+				// Wait for TreeSize to update.
+				if err := c.waitForRootUpdate(ctx); err != nil {
+					return err
+				}
+			default:
+				return err
+			}
```

Comment:
I like the more explicit set of return cases -- much easier to follow.

One nitlet - is it possible to outdent this block by closing the `select` just after the `default` arm?